### DISCUSSION TITLE
Question: Datalab row select event

### DISCUSSION THREAD

**Discussion by jbliesener** [gephi/gephi#2373] [Ideas]
This one is for Eduardo:

## Expected Behavior
In a plugin, I'd like to be notified when a row (node or edge) is selected in the Data Laboratory or in the Graph Overview

## Current Behavior
While I can create a "Tool" object to be informed about node and edge selections in the Graph Overview, I can't find a related event listener in DataLab. I thought that DataTablesEventListener may be a good starting point, but the doc says in big bold letters "Only data table UI should be an implementation of this listener". 

## Possible Solution
Maybe implementing a generic node selection listener? Maybe it already exists and I just can't find it?

## Steps to Reproduce
N/A

## Context
My plugin manages additional data, apart from the published node attributes. Actually, this structured data is very diverse and requires a specific embedded database to be stored. I was looking for a way to execute an online query to this node-specific database and show the result in a separate window. In Graph Overview, I am already using the approach of implementing a "Tool", but now the requirement has arisen to show this additional data in datalab as well.

Basically, what I need is a way to be notified whenever a row has been selected in datalab and whenever the Visulization mode changes to and from Datalab mode. That would allow me to query the data from the database and show it in a separate TopComponent.

Any suggestion of how to do this?

## Your Environment

* Version used: Gephi 0.9.2
* Java version:  1.8
* Operating System:  Multi OS: Win/Lin/Macos
